src/core/router/history/base.js

```diff
@@ -26,7 +26,9 @@ function getFileName(path, ext) {
     ? path
     : /\/$/g.test(path)
     ? `${path}README${ext}`
-    : `${path}${ext}`;
+    : path.indexOf('?') === -1
+    ? `${path}${ext}`
+    : path;
```

Comment:
We need to handle all URLs the same with or without the query, just like regular HTML in a browser.

For example, `/path/to/folder/?foo=123` has a query, and we still need to look for `/path/to/folder/README.md?foo=123`, but this skips anything with a query.

In the end, we need to operate on the path separately, and always send the query with the final result.

Reply:
@trusktr, is there anything that could help push this along? I'm not that familiar with  js but I am good with regular expressions. If you can provide a little more context on the issue perhaps I could put a little time into it, as fixing this would open up a lot of doors for those of us needing to federate markdown files from private repos/orgs on GitHub.

Reply:
@vergilcw sorry for late reply. I think the solution should determine the fully qualified URL (f.e. `anchor.href = partialUrlString; const fullyQualifiedUrl = anchor.href`) and then use a `new URL` object to get all the pieces and use them correctly.